packages/components/text/src/text.tsx

```diff
@@ -154,6 +154,7 @@ export type TBodyProps = {
   as?: 'span' | 'p';
```

Comment:
I think I see what you mean. Good point👍.

Reply:
I don't have a strong opinion on it, but I can totally see the point. And I can do it however we see it's the best fit, so let me know how we proceed.

> nit: I understand that the main purpose here is to replace `<s className={styles.removed}>{value}</s>`. I can see we also have a `<Strike>` component rendering as `<del>` in certain parts of the MC. Perhaps it might be useful to also add an option to render `<Body>` and `<Detail>` as `<del>` to convey special semantic meaning in some cases and be able to use this component there. I don't have any strong feelings this is the right thing to do here though 🤷‍♂️ Just wondering what others think

Reply:
@kark how do we proceed with this PR? Should we go into more details about your suggestion or should I merge this?
I've seen in few places in MC that `<b>` tag was used, instead of passing the `isBold` prop. Then I suppose if we change `isStrikethrough` to `<del>`, we should also take care of the bold case to be consistent, right? Or did you had in mind that both tag and prop should be used?

Reply:
Thanks for your thoughts. To my memory the MC internally only uses `<b>`-tags inside of translated messages whenever the translated string contains the formatting requirement. It's twofold however. Assume have the message `Hi <b>Klaudija</b>`. 

We can define a 

```js
const getBold = (msg) => (
  <Text.Body as="span" isBold>
    {msg}
  </Text.Body>
);
```

We can then pass it into an `intlMessage` as `values`

```js
<Text.Body
    intlMessage={{
        ...messages.fooMsg,
        values: {
        b: getBold,
        },
    }}
/>
```

Inside of our message `<b>Klaudija</b>` will then become `<Text.Body as="span" isBold>Klaudja</Text.Body>` which in turns is actually rendered. This in turn yields [this](https://github.com/commercetools/ui-kit/blob/main/packages/components/text/src/text.styles.ts#L23https://github.com/commercetools/ui-kit/blob/main/packages/components/text/src/text.styles.ts#L23). 

There might be cases where the `<b>`-tag inside a message actually renders an element as `react-intl` didn't always support us passing in a React component. However, those cases should be cleaned up. I'll take note and investigate it.

So given that it appears that using a `<b>`-tag inside the MC would render it inside a translated message is not correct. Cases where `b`-tags are used and don't "redirect" to `Text.Body` should be refactored ideally. Given that a `isBold` doesn't render a `<b>`-tag in ui-kit but styles using css means for me that this instance also doesn't have to render a `del`-tag. I like @kark point though. As using thee tags are implementation detail to ui-kit we could change this separately I think if we want to.

Reply:
I've seen it in for example `cart-summary-section.jsx` and it got me puzzled then why we don't render `<b>` tag instead we pass a prop. But thank you @tdeekens for the explanation! 

```
<Text.Body
    intlMessage={{
        ...messages.customer,
        values: {
             name: (
                  <b>
                       {formatCustomerName(customerFetcher.customer)}
                   </b>
             ),
             email: customerFetcher.customer.email,
        },
    }}
 />
```

Reply:
Yeah all these cases are flavours of wanting the same. I think cases where

1. `b`-tags are used normally outside of translations should become Text with isBold
2. `b`-tags used inside translations but not routing to Text isBold 

Both should be refactored ideally.